packages/node/src/indexer/indexer.manager.ts

```diff
@@ -169,30 +178,128 @@ export class IndexerManager {
     }
   }
 
-  private getStartBlockFromDataSources() {
-    const startBlocksList = this.getDataSourcesForSpecName().map(
-      (item) => item.startBlock ?? 1,
-    );
-    if (startBlocksList.length === 0) {
-      logger.error(
-        `Failed to find a valid datasource, Please check your endpoint if specName filter is used.`,
+  private async ensureProject(): Promise<string> {
+    let schema = await this.getProjectSchema();
+    if (!schema) {
+      schema = await this.createProjectSchema();
+    } else {
+      if (argv['force-clean']) {
+        try {
+          // drop existing project schema and metadata table
+          this.sequelize.dropSchema(schema, {
+            logging: false,
+            benchmark: false,
+          });
+
+          // remove schema from subquery table (might not exist)
+          await this.sequelize.query(
+            ` DELETE
+              FROM public.subqueries
+              where db_schema = :subquerySchema`,
+            {
+              replacements: { subquerySchema: schema },
+              type: QueryTypes.DELETE,
+            },
+          );
+
+          logger.info('force cleaned schema and tables');
+        } catch (err) {
+          logger.error(err, 'failed to force clean schema and tables');
+        }
+        schema = await this.createProjectSchema();
+      }
+    }
+    return schema;
+  }
+
+  private async getProjectSchema(): Promise<string> {
+    let schema = argv.schema;
+    if (!schema) {
+      schema = this.nodeConfig.subqueryName;
+    }
+    const schemas = (await this.sequelize.showAllSchemas(
+      undefined,
```

Comment:
```suggestion
    this.sequelize.showAllSchemas()
```

Reply:
argument for logging in this function isn't optional, explicit undefined required